data/basal/sample_builder.go

```diff
@@ -0,0 +1,56 @@
+package basal
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/mdblp/tide-whisperer-v2/v2/schema"
+	schemaV1 "github.com/tidepool-org/tide-whisperer/schema"
+)
+
+func makeBasalSample(startTime time.Time, endTime time.Time, deliveryType string, sample schema.BasalSample) schema.BasalSample {
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches to further communicate your intent.  -->
**🎯 suggestion (non-blocking):** ‏ since this is already returning an error, can we check startTime < endTime ?

Reply:
I will check that and swap start and end time on that case